pkg/migration/file.go

```diff
@@ -26,7 +26,10 @@ type MigrationFile struct {
 	Statements []string
 }
 
-var migrateFilePattern = regexp.MustCompile(`^([0-9]+)_(.*)\.sql$`)
+var (
+	migrateFilePattern = regexp.MustCompile(`^([0-9]+)_(.*)\.sql$`)
+	typeNamePattern    = regexp.MustCompile(`type "([^"]+)" does not exist`)
```

Comment:
We should distinguish type not found errors where the schema is already specified. For eg.

```bash
postgres=> CREATE TABLE test (path extensions.ltree NOT NULL);
ERROR:  type "extensions.ltree" does not exist
LINE 1: CREATE TABLE test (path extensions.ltree NOT NULL);
```

In this case, user may not have enabled the `ltree` extension or the extension might be installed in a different schema. So our hint should not suggest using schema qualified reference.

Reply:
probably better to define a new method `func IsSchemaQualified(typename string) bool` that accepts the value returned by `extractTypeName`.